chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -0,0 +1,499 @@
+			if (!testMode)
+			{
+				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants.identifier);
+			}
+		}
+	}
+
+	/**
+	 * Updates ReferenceEntities that refer to ReferenceMolecules. This method will
+	 * ensure that the "name" array of a ReferenceEntity associated with
+	 * <code>molecule</code> has <code>chebiName</code> as the <em>first</em> name
+	 * in its list of names.
+	 * 
+	 * @param molecule - a ReferenceMolecule. ReferenceEntities that reference this ReferenceMolecule will be updated.
+	 * @param chebiName - the name from ChEBI. Should be the first name in the "name" array for any updated ReferenceEntity.
+	 * @param instanceEdit - and InstanceEdit that the changes will be associated with.
+	 * @throws Exception
+	 * @throws InvalidAttributeException
+	 * @throws InvalidAttributeValueException
+	 */
+	private void updateReferenceEntities(GKInstance molecule, String chebiName, GKInstance instanceEdit) throws Exception, InvalidAttributeException, InvalidAttributeValueException
+	{
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> refEntities = (Collection<GKInstance>) molecule.getReferers("referenceEntity");
+
+		for (GKInstance refEntity : refEntities)
+		{
+			@SuppressWarnings("unchecked")
+			LinkedList<String> names = new LinkedList<String>(refEntity.getAttributeValuesList("name"));
+			// Now we must ensure that the name from the ChEBI molecule is the FIRST name in
+			// the referenceEntity's list of names.
+			if (!names.isEmpty())
+			{
+				// name[0] does not match - but before adding chebName, ensure that it's not
+				// already somewhere else in the array.
+				if (!names.get(0).toLowerCase().equals(chebiName.toLowerCase()))
+				{
+					int i = 0;
+					boolean nameFound = false;
+					while (!nameFound && i < names.size())
+					{
+						String name = names.get(i);
+						// We found the chebi name so now we swap.
+						if (name.toLowerCase().equals(chebiName.toLowerCase()))
+						{
+							nameFound = true;
+							// Remove the ChEBI name at position i
+							names.remove(i);
+							// Add the ChEBI name back at position 0.
+							names.add(0, chebiName);
+						}
+						i++;
+					}
+					// If we went through the whole array and didn't find the ChEBI name,
+					// we must add it, at the begining of the list of names.
+					if (!nameFound)
+					{
+						names.add(0, chebiName);
+					}
+				}
+			}
+			// Update the names of the ReferenceEntity.
+			refEntity.setAttributeValue(ReactomeJavaConstants.name, names);
+			if (!testMode)
+			{
+				addInstanceEditToExistingModifieds(instanceEdit, refEntity);
+				
+				adaptor.updateInstanceAttribute(refEntity, ReactomeJavaConstants.name);
+			}
+		}
+	}
+
+	private void addInstanceEditToExistingModifieds(GKInstance instanceEdit, GKInstance instance) throws InvalidAttributeException, Exception, InvalidAttributeValueException
```

Comment:
@cookersjs You might want to check your code to ensure you're adding a new InstanceEdit to the existing list, rather than overwriting the existing list of "modified" InstanceEdits. I'd forgotten this "gotcha!" until a few minutes ago.

Reply:
I'd forgotten this potential problem, but I'd already dealt with in AddLinks: https://github.com/reactome/add-links/blob/de443d43852a9307c387daa9d3d1abbd5c36c4b1/src/main/java/org/reactome/addlinks/db/ReferenceCreator.java#L289-L292

Reply:
Thanks for the heads up! I copied the AddLinks Instance Edits, and it correctly pushed to the array